src/hooks/use-add-site.ts

```diff
@@ -31,6 +35,25 @@ export function useAddSite() {
 		[ sites ]
 	);
 
+	const handleCustomDomainChange = useCallback(
+		( value: string | null ) => {
+			setCustomDomain( value );
+			// Validate custom domain if enabled
+			const domainPattern =
+				/^[\p{L}\p{N}][\p{L}\p{N}-]{0,61}[\p{L}\p{N}](?:\.[\p{L}\p{N}-]{2,})+$/u;
```

Comment:
```suggestion
				/^[a-z0-9][a-z0-9-]{0,61}[a-z0-9](\.[a-z0-9][a-z0-9-]{0,61}[a-z0-9])+$/i;
```
I don't think it's correct to allow any unicode characters in the domain name.

I believe my suggestion catches the relevant syntax criteria: two or more labels that can only contain ASCII letters (lower- or uppercase), digits, and hyphens (but not start or end with a hyphen).

Reply:
We need to allow for more than ASCII letters, digits, and hyphens at this step, to support other supported languages. The domain is now encoded using `domainToASCII` when we start proxy or write to the hosts file.

Reply:
Or maybe I'm misunderstanding how we intend to support internationalized domain names..?

Reply:
Thanks for the clarification, @wojtekn 👍

Reply:
```suggestion
				/^[\p{L}\p{N}][\p{L}\p{N}-]{0,61}[\p{L}\p{N}](?:\.[\p{L}\p{N}][\p{L}\p{N}-]{0,61}[\p{L}\p{N}])+$/u;
```
Then let's do something like this to account for my previous comment about hyphens in all labels.

Reply:
@fredrikekelund This is already accounted for in my tests, can you give me an example of domain that don't pass validation?

Reply:
I see now that hyphens are already allowed in all labels, but we still need to adjust the regex to account for the fact that no label can start or end with hyphens. For example, `hey.-hey` passes the current regex, which is incorrect.

Reply:
Should be addressed in d6a42be69d3bb66a6b206733a272191fc42042a0